src/daml/Daml/Trading/Account/Account.daml

```diff
@@ -0,0 +1,254 @@
+import Daml.Finance.Interface.Types.Common (AccountKey(..), Id(..))
+import Daml.Trading.Reward.Badge qualified as Badge
+import DA.Map qualified as M
+import Daml.Finance.Interface.Holding.Util (getAmount)
+import Daml.Finance.Interface.Holding.Base qualified as Holding (I)
+import Daml.Finance.Interface.Holding.Fungible qualified as Fungible (I, Split(..), SplitResult(..),Merge(..))
+import DA.List (head,tail)
+
+cashLabel = "Cash"
+nftLabel = "Nft"
+
+-- to fetch the account factory for subsequent account creation
+template AccountFactoryReference 
+  with 
+    provider: Party 
+    accountFactoryCid: ContractId Account.F 
+  where 
+    signatory provider 
+    key provider: Party 
+    maintainer key 
+
+template HoldingFactoryReference 
+  with 
+    provider: Party 
+    holdingType: Text 
+    holdingFactoryCid: ContractId Holding.F 
+  where 
+    signatory provider 
+    key (provider, holdingType): (Party, Text) 
+    maintainer key._1 
+
+template InstrumentFactoryReference 
+  with 
+    provider: Party 
+    instrumentType: Text 
+    instrumentFactoryCid: ContractId Instrument.F 
+  where 
+    signatory provider 
+    key (provider, instrumentType): (Party, Text) 
+    maintainer key._1 
+
+template Request 
+  with 
+    custodian: Party
+    owner: Party 
+    accountType: AccountType 
+    username: Text 
+    realName: Text 
+    passportNumber: Text 
+    nationalId: Text 
+    address: Text 
+  where 
+    signatory owner 
+    observer custodian 
+
+    choice Request_Accept: ContractId Account 
+      controller custodian 
+      do
+        create Account 
+          with 
+            custodian 
+            owner 
+            accountType 
+            username 
+            realName 
+            passportNumber 
+            nationalId 
+            address 
+            optCashAccount = None 
+            optNftAccount = None 
+            rewardPoints = 0 
+            badges = []
+
+    choice Request_Decline : ()
+      controller custodian
+      do pure ()
+
+    choice Request_Withdraw : ()
+      controller owner
+      do pure ()
+
+template Account 
+  with 
+    custodian: Party 
+    owner: Party 
+    accountType: AccountType 
+    username: Text 
+    realName: Text 
+    passportNumber: Text 
+    nationalId: Text 
+    address: Text 
+    optCashAccount: Optional (ContractId Account.F, ContractId Holding.F, AccountKey) 
+    optNftAccount: Optional (ContractId Account.F, ContractId Holding.F, ContractId Instrument.F, AccountKey) 
+    rewardPoints : Int
+    badges : [ContractId Badge.Badge]
+  where 
+    signatory custodian 
+    observer owner
+    key (custodian, owner): (Party, Party)
+    maintainer key._1 
+
+    choice Update_Permission: ContractId Account 
+      with 
+        newAccountType: AccountType 
+      controller custodian 
+      do 
+        assertMsg ("New account type should be different from current one") (accountType /= newAccountType)
```

Comment:
you can get rid off unnecessary parenthesis, familiarise yourself with operators like $ that make code cleaner, it's not really an issue rather code style